src/jit/namedintrinsiclist.h

```diff
@@ -16,10 +16,84 @@ enum NamedIntrinsic : unsigned int
     NI_System_Collections_Generic_EqualityComparer_get_Default = 4,
 #if FEATURE_HW_INTRINSICS
     NI_HW_INTRINSIC_START,
-#define HARDWARE_INTRINSIC(id, name, isa) NI_##id,
+#define HARDWARE_INTRINSIC(id, name, isa, ival, size, numarg, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, category)       \
+    NI_##id,
 #include "hwintrinsiclistxarch.h"
     NI_HW_INTRINSIC_END
 #endif
 };
 
+#if FEATURE_HW_INTRINSICS && defined(_TARGET_XARCH_)
+enum HWIntrinsicCategory : unsigned int
```

Comment:
I still think this would be simpler by having 2-3 distinct fields, rather than a bitmask.

Reply:
Right now, it's really hard to tell, "what is a category" and "what is a flag".

So it isn't easy to see something like "I can be `SimpleSIMD`" (because its a category), but "I can also be `Commutative` (because its a flag).

I think we should also differentiate between `SIMD`, `Scalar`, and `SIMDScalar` (as the handling between them are sometimes different).

Reply:
> I think we should also differentiate between SIMD, Scalar, and SIMDScalar (as the handling between them are sometimes different).

I think `SIMDScalar` is same to `SIMD` in this table-driven framework.

Reply:
I think we want a flag (or some mechanism) to differentiate them, since SIMDScalar and SIMD have different "rules" and exception handling

Reply:
Okay, we can add flags at anytime when we need.

Reply:
> Each instruction should have a single category, but could have multiple flags.

Hmm, the current categories all look like "flags", can we just change the name to `HW_FLAG_XXX`?

Reply:
I'm not sure that's the right solution. I would have thought the following would be fundamental categories that would be mutually exclusive: SimpleSIMD, Scalar, MemoryLoad, MemoryStore and NoCodeGen (and perhaps others).
However, I would think that these would be flags/properties: FullRangeIMM, Generic, UnfixedSIMDSize, Commutative.

Reply:
Sounds good, will change.